docs/Containers/Mosquitto.md

```diff
@@ -144,7 +159,7 @@ You *may* see the same pattern in Portainer, which reports the *base image* as "
 
 > Whether you see one or two rows depends on the version of `docker-compose` you are using and how your version of `docker-compose` builds local images.
 
-### Migration considerations
+### <a name="migration-considerations"> Migration considerations </a>
```

Comment:
Bookmarks are automatically generated as id-attributes, no need to add them manually using the deprecated name-attribute.

![image](https://user-images.githubusercontent.com/95980324/159694980-204e66cc-6f9a-4bd5-8042-26ebca9f745b.png)

Also, using `<a>` creates blue "link" headings, that on a dark background are a bit low on contrast and may make users think this it's a link to be clicked:
![image](https://user-images.githubusercontent.com/95980324/159696157-9f8f14d1-bb41-44b4-8a8c-7666e236b5f3.png)

Reply:
Believe me, I'm no fan of a mixture of blue and black headings and what I prefer to do in most cases is step through and put explicit anchors on all titles so that the rendering is consistent. I mostly do that systematically on "new" or "wholesale rewrites" rather than "just because I'm touching a document". I did think about doing it for this document but decided to leave it for another time.

Why do I put explicit anchors on things rather than rely on auto-generation? Because auto-generation is fragile (a typo can break a link) and the internal page links aren't testable (at least, not in the tools I use - MacDown to HTML to BBEdit to "check document links"). Do you know of another way to test link integrity?

Also, the blue anchors seem to be a feature of mkdocs. Using a snippet from the Node-RED doc as the example - raw:

![BFFF220B-6728-4249-AD7B-E211433BEB57](https://user-images.githubusercontent.com/34226495/159801000-9f905ec6-502e-4d37-8a0b-00b445379d97.jpeg)

As rendered by GitHub:

![91B71143-342E-4818-B397-F7D8555FE1B4](https://user-images.githubusercontent.com/34226495/159801022-27faddf0-127b-4b7c-a5f0-0d13c28d6ee2.jpeg)

As rendered by Wiki:

![998B19E6-002A-4FF3-BB48-F77EF94A6FEB](https://user-images.githubusercontent.com/34226495/159801026-f394cdae-99e3-492e-b00d-873664c838de.jpeg)

It's clearly a mkdocs difference so, my guess is that the blue rendering might have a control somewhere that we haven't stumbled across.

> Seeing as we're on this topic, I'm no fan of the way the "Variable definitions" title is rendered in the Wiki. I'd prefer bold rather than a slightly larger font size.

Last thing. My screen shots above were taken on an iPad running in 24*7 Dark Mode. I don't see any difference when I turn off Dark Mode so there's likely also a platform variation too.

Reply:
Also, there are quite a few...

![770859C7-0654-4665-ADDE-5987E5377491](https://user-images.githubusercontent.com/34226495/159804544-18724d05-d902-4fb0-9b71-c9cf2affcda2.jpeg)

Reply:
hmm, yes apparently there's still some around. But can't find nearly that many, did you forget to git fetch?
```
pi@raspberrypi:~/IOTstack $ git fetch -q upstream && git grep '<a name=' upstream/master -- | cat
upstream/master:README.md:#### <a name="networkHandEdit"> adopting networking changes by hand </a>
upstream/master:docs/Containers/NextCloud.md:## <a name="untrustedDomain">"Access through untrusted domain"</a>
upstream/master:docs/Containers/NextCloud.md:## <a name="security"> Security considerations</a>
upstream/master:docs/Containers/Node-RED.md:#### <a name="sshHostAddr">«HOSTADDR» (required)</a>
upstream/master:docs/Containers/Node-RED.md:#### <a name="sshUserID">«USERID» (required)</a>
upstream/master:docs/Containers/Node-RED.md:### <a name="sshStep2">Step 2: *Exchange keys with target hosts* (once per target host)</a>
upstream/master:docs/Containers/Node-RED.md:<a name="applyDockerfileChange">You apply a Dockerfile change like this:</a>
upstream/master:docs/Containers/WireGuard.md:#### <a name="configureAlways">Fields that you should always configure </a>
upstream/master:docs/Containers/WireGuard.md:* `PEERDNS=auto` with `custom-cont-init` <a name="customContInit"></a>
```

Can your editor validate e.g.
```md
## Configuration <span id="conf"></span>                                          
```

Reply:
Or as [someone on stackoverflow recommends](https://stackoverflow.com/questions/5319754/cross-reference-named-anchor-in-markdown/7335259#7335259):
```md
## <a name="conf"></a>Configuration 
```
No need to have the header text inside the tag, right?

Reply:
Actually, it's worse than that. I was in my "development" branch. Shoulda switched to "master". 🤬

To answer your second question, the "stackoverflow" suggestion works. I didn't try the `span`. This time I was smart enough to be in master branch with a pull reporting up-to-date before I started. The basic process I follow is to open the `.md` in MacDown and use its "Export to HTML", then open the `.html` in BBEdit and use its "Check document links" command.

Right out-of-the-box:

<img width="608" alt="Node-RED-before" src="https://user-images.githubusercontent.com/34226495/159845737-3e76de64-b925-4105-8242-23a3fa99edd7.png">

Anything referring to an image can be ignored - I was working with the files on my desktop. The others are "real" errors. Using the first as an example, it's referring to "adding-extra-nodes" but the title has changed to "Adding extra packages" so the dynamically-generated anchor doesn't work.  

Then I worked through and changed all the remaining "traditional" anchors into stackoverflow form. The diff is:

```
$ diff Node-RED-before.md Node-RED-after.md 
481c481
< #### <a name="sshHostAddr">«HOSTADDR» (required)</a>
---
> #### <a name="sshHostAddr"></a>«HOSTADDR» (required)
510c510
< #### <a name="sshUserID">«USERID» (required)</a>
---
> #### <a name="sshUserID"></a>«USERID» (required)
534c534
< ### <a name="sshStep2">Step 2: *Exchange keys with target hosts* (once per target host)</a>
---
> ### <a name="sshStep2"></a>Step 2: *Exchange keys with target hosts* (once per target host)
874c874
< <a name="applyDockerfileChange">You apply a Dockerfile change like this:</a>
---
> <a name="applyDockerfileChange"></a>You apply a Dockerfile change like this:
```

Then I re-ran the test:

<img width="608" alt="Node-RED-after" src="https://user-images.githubusercontent.com/34226495/159845804-3c86327c-8d0a-4462-b7eb-2279c9c246d7.png">

No new errors. Clicking the sources of the newly *stackoverflowified* links show they still get to their destination, while the destination anchors are non-blue (which was the goal).

Meanwhile, the other errors BBEdit found mean:

<img width="743" alt="Screen Shot 2022-03-24 at 16 06 41" src="https://user-images.githubusercontent.com/34226495/159846358-609d92bc-87a0-434f-848f-fbaaa250edd8.png">

Red is dead. Green is clean. 🤮

We have some work to do. Sigh.

Reply:
I've changed the links to stackoverflow style (plus one fix-up).